waspc/cli/tests/Wasp/Cli/Command/News/ListingTest.hs

```diff
@@ -0,0 +1,101 @@
+module Wasp.Cli.Command.News.ListingTest where
+
+import Data.List (subsequences)
+import qualified Data.Time as T
+import Test.Hspec
+import Test.Hspec.QuickCheck (prop)
+import Test.QuickCheck (Gen, elements, forAll, (===), (==>))
+import Wasp.Cli.Command.News.Core (NewsEntry (..), NewsLevel (..))
+import Wasp.Cli.Command.News.Listing
+  ( NewsListing (..),
+    getNewsFromListing,
+    getNewsToMarkAsSeen,
+    getNewsToShow,
+    isConfirmationRequired,
+  )
+import Wasp.Cli.Command.News.LocalNewsState
+  ( LocalNewsState,
+    emptyLocalNewsState,
+    markNewsAsSeen,
+  )
+
+spec_getNewsToShow :: Spec
+spec_getNewsToShow = do
+  describe "getNewsToShow" $ do
+    describe "for UserRequestedAllNews" $ do
+      it "returns all news in the listing" $ do
+        let newsState = markNewsAsSeen allNews emptyLocalNewsState
+        getNewsToShow newsState (UserRequestedAllNews allNews) `shouldBe` allNews
+
+    describe "for WaspRequestedMustSeeNews" $ do
+      it "returns unseen news that are at least important" $ do
+        let newsState = markNewsAsSeen [info1, important1, critical1] emptyLocalNewsState
+        getNewsToShow newsState (WaspRequestedMustSeeNews [info1, important1, important2, critical1, critical2])
+          `shouldMatchList` [important2, critical2]
+
+spec_isConfirmationRequired :: Spec
+spec_isConfirmationRequired = do
+  describe "isConfirmationRequired" $ do
+    describe "for UserRequestedAllNews" $ do
+      it "returns False" $ do
+        let newsState = markNewsAsSeen allNews emptyLocalNewsState
+        isConfirmationRequired newsState (UserRequestedAllNews allNews) `shouldBe` False
```

Comment:
Again you are testing on an edge case where all the news are seen. I know you know that for the implementation it doesn't matter, but that is an implemenation detail.